### ISSUE TITLE
Combining an input sequence with hyperop & regex can break SPESH

### ISSUE THREAD

**Issue by dmaestro** [MoarVM/MoarVM#775]
Where test_input is a file of 151 lines:
`"project_gbclient_t_controller_signup_process_purchase_validation_07-wordpress_blog_title_t"`

This script that strips the quotes fails non-deterministicly:
```
its-MacBook-Pro-DLS:~ douglasschrag$ cat test_input test_input | perl6 -e 'say $*IN.lines.head(250).grep({  / ^ <[ " ]> <[ \w - ]>+ <[ " ]> $ / })».subst( / \" /, "", :g ).elems;'
239
its-MacBook-Pro-DLS:~ douglasschrag$ cat test_input test_input | perl6 -e 'say $*IN.lines.head(250).grep({  / ^ <[ " ]> <[ \w - ]>+ <[ " ]> $ / })».subst( / \" /, "", :g ).elems;'
198
its-MacBook-Pro-DLS:~ douglasschrag$ cat test_input test_input | MVM_SPESH_DISABLE= perl6 -e 'say $*IN.lines.head(250).grep({  / ^ <[ " ]> <[ \w - ]>+ <[ " ]> $ / })».subst( / \" /, "", :g ).elems;'
232
its-MacBook-Pro-DLS:~ douglasschrag$ cat test_input test_input | MVM_SPESH_DISABLE= perl6 -e 'say $*IN.lines.head(250).grep({  / ^ <[ " ]> <[ \w - ]>+ <[ " ]> $ / })».subst( / \" /, "", :g ).elems;'
242
its-MacBook-Pro-DLS:~ douglasschrag$ cat test_input test_input | MVM_SPESH_DISABLE=1 perl6 -e 'say $*IN.lines.head(250).grep({  / ^ <[ " ]> <[ \w - ]>+ <[ " ]> $ / })».subst( / \" /, "", :g ).elems;'
250
its-MacBook-Pro-DLS:~ douglasschrag$ cat test_input test_input | MVM_SPESH_DISABLE=1 perl6 -e 'say $*IN.lines.head(250).grep({  / ^ <[ " ]> <[ \w - ]>+ <[ " ]> $ / })».subst( / \" /, "", :g ).elems;'
250
its-MacBook-Pro-DLS:~ douglasschrag$ cat test_input test_input | MVM_SPESH_DISABLE=1 perl6 -e 'say $*IN.lines.head(250).grep({  / ^ <[ " ]> <[ \w - ]>+ <[ " ]> $ / })».subst( / \" /, "", :g ).elems;'
250
its-MacBook-Pro-DLS:~ douglasschrag$ cat test_input test_input | MVM_SPESH_DISABLE=1 perl6 -e 'say $*IN.lines.head(250).grep({  / ^ <[ " ]> <[ \w - ]>+ <[ " ]> $ / })».subst( / \" /, "", :g ).elems;'
250
its-MacBook-Pro-DLS:~ douglasschrag$ perl6 -v
This is Rakudo version 2017.12-97-gf8d8481 built on MoarVM version 2017.12-19-g40a79ee
implementing Perl 6.c.
```
I tried this first with `».EVAL` but modified the script to eliminate that - still breaks.

The above is on Mac OS X 10.11.6

Similar results on CentOS 6.6:
```
[dschrag@dschrag ~]$ cat test_input test_input | MVM_SPESH_DISABLE= perl6 -e 'say $*IN.lines.head(250).List.grep({  / ^ <[ " ]> <[ \w - ]>+ <[ " ]> $ / })».subst( / \" /, "", :g ).elems;'
172
[dschrag@dschrag ~]$ cat test_input test_input | MVM_SPESH_DISABLE= perl6 -e 'say $*IN.lines.head(250).List.grep({  / ^ <[ " ]> <[ \w - ]>+ <[ " ]> $ / })».subst( / \" /, "", :g ).elems;'
171
[dschrag@dschrag ~]$ cat test_input test_input | MVM_SPESH_DISABLE=1 perl6 -e 'say $*IN.lines.head(250).List.grep({  / ^ <[ " ]> <[ \w - ]>+ <[ " ]> $ / })».subst( / \" /, "", :g ).elems;'
250
[dschrag@dschrag ~]$ cat test_input test_input | MVM_SPESH_DISABLE=1 perl6 -e 'say $*IN.lines.head(250).List.grep({  / ^ <[ " ]> <[ \w - ]>+ <[ " ]> $ / })».subst( / \" /, "", :g ).elems;'
250
[dschrag@dschrag ~]$ perl6 -v
This is Rakudo version 2017.10 built on MoarVM version 2017.10
implementing Perl 6.c.
```

**Comment by MattOates** [comment]
I managed to reproduce this locally though had to bump up the number of lines a little bit. I'm on a different version of macOS 10.12.6 this is with Rakudo version 2017.12-97-gf8d8481c9 built on MoarVM version 2017.12-15-g8414888be

**Comment by jnthn** [comment]
For what it's worth, to get more deterministic failures, setting `MVM_SPESH_BLOCKING=1` in the environment usually does the trick.

**Comment by dogbert17** [comment]
For me MVM_SPESH_INLINE_DISABLE=1 seems to resolve the problem

**Comment by dmaestro** [comment]
New test with Rakudo* 2018.01 (linux)
Pretty much the same results:
```
doug@ender$ MVM_SPESH_BLOCKING=1 perl6 spesh_fail.t test_input test_input
118
doug@ender$ MVM_SPESH_BLOCKING=1 perl6 spesh_fail.t test_input test_input
118
doug@ender$ MVM_SPESH_BLOCKING= perl6 spesh_fail.t test_input test_input
184
doug@ender$ MVM_SPESH_BLOCKING= perl6 spesh_fail.t test_input test_input
185
doug@ender$ MVM_SPESH_INLINE_DISABLE=1 perl6 spesh_fail.t test_input test_input
302
doug@ender$ MVM_SPESH_INLINE_DISABLE=1 perl6 spesh_fail.t test_input test_input
302
doug@ender$ MVM_SPESH_INLINE_DISABLE=1 perl6 spesh_fail.t test_input test_input
302
doug@ender$ MVM_SPESH_INLINE_DISABLE=1 perl6 spesh_fail.t test_input test_input
302
doug@ender$ MVM_SPESH_OSR_DISABLE=1 perl6 spesh_fail.t test_input test_input
276
doug@ender$ MVM_SPESH_OSR_DISABLE=1 perl6 spesh_fail.t test_input test_input
260
doug@ender$ MVM_SPESH_OSR_DISABLE=1 perl6 spesh_fail.t test_input test_input
263
```
Removed `.head()`
```
# spesh_fail.t
say $*ARGFILES.lines.grep({ / ^ <[ " ]> <[ \w - ]>+ <[ " ]> $ / })\
    ».subst( / \" /, "", :g ).elems;
```

Does appear to be most directly related to inlining.

**Comment by dmaestro** [comment]
And *both examples are fixed* in 2018.09 !!
```
DLS-Mac-mini:study doug$ wc -l M775_test_input
     192 M775_test_input
DLS-Mac-mini:study doug$ cat M775_test_input | /usr/local/bin/perl6 -e 'say $*IN.lines.head(250).grep({  / ^ <[ " ]> <[ \w - ]>+ <[ " ]> $ / })».subst( / \" /, "", :g ).elems;'
192
DLS-Mac-mini:study doug$ cat M775_test_input | /usr/local/bin/perl6 -e 'say $*IN.lines.head(250).grep({  / ^ <[ " ]> <[ \w - ]>+ <[ " ]> $ / })».subst( / \" /, "", :g ).elems;'
192
```
```
DLS-mini:study doug$ /usr/local/bin/perl6 -ne '.say if /">" \w* [(.)$0]**3 \w* "<"/' words.html
        <td id="LC11085" class="blob-code blob-code-inner js-file-line">bookkeeper</td>
        <td id="LC11086" class="blob-code blob-code-inner js-file-line">bookkeepers</td>
        <td id="LC11087" class="blob-code blob-code-inner js-file-line">bookkeeping</td>
        <td id="LC11088" class="blob-code blob-code-inner js-file-line">bookkeepings</td>
DLS-Mac-mini:study doug$ /usr/local/bin/perl6 -ne '.say if /">" \w* [(.)$0]**3 \w* "<"/' words.html
        <td id="LC11085" class="blob-code blob-code-inner js-file-line">bookkeeper</td>
        <td id="LC11086" class="blob-code blob-code-inner js-file-line">bookkeepers</td>
        <td id="LC11087" class="blob-code blob-code-inner js-file-line">bookkeeping</td>
        <td id="LC11088" class="blob-code blob-code-inner js-file-line">bookkeepings</td>
```
```
DLS-Mac-mini:study doug$ /usr/local/bin/perl6 -v
This is Rakudo version 2018.09 built on MoarVM version 2018.09
implementing Perl 6.c.
```
Thanks to @jnthn and others for the outstanding work!
